### ISSUE TITLE
parallelism: 1 to run sequential loop does not work. -> Example: parallelism-nested-workflow does not work.

### ISSUE THREAD

**Issue by se-jaeger** [argoproj/argo#1303]
**Is this a BUG REPORT or FEATURE REQUEST?**: BUG REPORT

**What happened**:
I ran the example: parallelism-nested-workflow like:
`argo submit https://raw.githubusercontent.com/argoproj/argo/master/examples/parallelism-nested-workflow.yaml`

**What you expected to happen**:
That the workflow runs at leas 120 seconds.
Because 4 steps runs sequential and sleep for 30 seconds. But just needs 37 seconds (see below)

**How to reproduce it (as minimally and precisely as possible)**:
Install Argo and start parallelism-nested-workflow as described above.

**Anything else we need to know?**:

**Environment**:
- Argo version:
```
$ argo version
argo: v2.2.1
  BuildDate: 2018-10-11T16:25:59Z
  GitCommit: 3b52b26190163d1f72f3aef1a39f9f291378dafb
  GitTreeState: clean
  GitTag: v2.2.1
  GoVersion: go1.10.3
  Compiler: gc
  Platform: darwin/amd64
```
- Kubernetes version :
```
$ kubectl version -o yaml
clientVersion:
  buildDate: 2018-04-27T09:22:21Z
  compiler: gc
  gitCommit: 81753b10df112992bf51bbc2c2f85208aad78335
  gitTreeState: clean
  gitVersion: v1.10.2
  goVersion: go1.9.3
  major: "1"
  minor: "10"
  platform: darwin/amd64
serverVersion:
  buildDate: 2018-06-21T11:34:22Z
  compiler: gc
  gitCommit: 32ac1c9073b132b8ba18aa830f46b77dcceb0723
  gitTreeState: clean
  gitVersion: v1.10.5
  goVersion: go1.9.3
  major: "1"
  minor: "10"
  platform: linux/amd64
```

**Other debugging information (if applicable)**:
- workflow result:
```
$ argo get parallelism-nested-workflow-dd4t9
Name:                parallelism-nested-workflow-dd4t9
Namespace:           argo
ServiceAccount:      default
Status:              Succeeded
Created:             Wed Apr 03 10:52:55 +0200 (1 minute ago)
Started:             Wed Apr 03 10:52:55 +0200 (1 minute ago)
Finished:            Wed Apr 03 10:53:32 +0200 (47 seconds ago)
Duration:            37 seconds
Parameters:
  seq-list:          ["a","b","c","d"]


STEP                                  PODNAME                                       DURATION  MESSAGE
 ✔ parallelism-nested-workflow-dd4t9
 └-·-✔ seq-step(0:a)
   | └-·-✔ jobs(0:1)                  parallelism-nested-workflow-dd4t9-638376637   35s
   |   └-✔ jobs(1:2)                  parallelism-nested-workflow-dd4t9-944691679   35s
   ├-✔ seq-step(1:b)
   | └-·-✔ jobs(0:1)                  parallelism-nested-workflow-dd4t9-172209403   34s
   |   └-✔ jobs(1:2)                  parallelism-nested-workflow-dd4t9-3950373625  35s
   ├-✔ seq-step(2:c)
   | └-·-✔ jobs(0:1)                  parallelism-nested-workflow-dd4t9-1234698765  35s
   |   └-✔ jobs(1:2)                  parallelism-nested-workflow-dd4t9-3688549039  36s
   └-✔ seq-step(3:d)
     └-·-✔ jobs(0:1)                  parallelism-nested-workflow-dd4t9-3817084031  34s
       └-✔ jobs(1:2)                  parallelism-nested-workflow-dd4t9-3068806957  34s
```

**Comment by jessesuen** [comment]
This was implemented in https://github.com/argoproj/argo/pull/1046 which is available in v2.3 only.

**Comment by se-jaeger** [comment]
Yes, I found this PR. I wasn't aware that this feature would be available in a future release because the example workflow already existed.
Thank you!
